### ISSUE TITLE
[Bug] Ray dashboard crashed

### ISSUE THREAD

**Issue by Murphy-OrangeMud** [ray-project/ray#18889]
### Search before asking

- [X] I searched the [issues](https://github.com/ray-project/ray/issues) and found no similar issues.

### Ray Component

Dashboard

### What happened + What you expected to happen

Ray dashboard crashed and didn't show anything.
I pressed F12 and find:

TypeError: Cannot read properties of undefined (reading 'length')
    at NodeInfo.tsx:211
    at Array.filter (<anonymous>)
    at Gn (NodeInfo.tsx:211)
    at Qi (react-dom.production.min.js:153)
    at _a (react-dom.production.min.js:175)
    at vl (react-dom.production.min.js:263)
    at cu (react-dom.production.min.js:246)
    at lu (react-dom.production.min.js:246)
    at Zl (react-dom.production.min.js:239)
    at react-dom.production.min.js:123

### Reproduction script

ray start --head --num-gpus 2 --dashboard-host '0.0.0.0'

### Are you willing to submit a PR?

- [ ] Yes I am willing to submit a PR!

**Comment by zhaichenhao1889** [comment]
I am having the same issue

**Comment by witt3rd** [comment]
Same issue on 1.9 WSL2 Ubuntu 20.4 --- I am surprised that this hasn't been addressed yet --- the dashboard is not usable

**Comment by yyzlong** [comment]
Got same issue on ray 1.9.0 under centos 7.7 & Python3.7.3. Dashboard worked if downgrade to ray 1.1.0 on the same machine.    It seems to me the dashboard is not stable and somehow broken after upgrading a newer version which adds experimental dashboard feature. Hopefully, someone can provide a workaround on this issue ASAP.

**Comment by newmanwang** [comment]
It happens after some worker node marked as dead as
```
The node with node id: 63049713526c2df2956d5f9296055ed9988cacd237c4c5d2aafe1a5a and ip: 192.168.11.21 has been marked dead because the detector has missed too many heartbeats from it. This can happen when a raylet crashes unexpectedly or has lagging heartbeats
```

chrome console log:
```
TypeError: Cannot read properties of undefined (reading 'length')
    at NodeInfo.tsx:211:52
    at Array.filter (<anonymous>)
    at _n (NodeInfo.tsx:211:30)
    at Xi (react-dom.production.min.js:153:146)
    at Go (react-dom.production.min.js:175:309)
    at ys (react-dom.production.min.js:263:406)
    at El (react-dom.production.min.js:246:265)
    at fl (react-dom.production.min.js:246:194)
    at sl (react-dom.production.min.js:239:172)
    at react-dom.production.min.js:123:115
```

**Comment by newmanwang** [comment]
After node marked as dead due to timeout, dashboard also crash
```
[2022-02-18 12:45:43,851 I 149 149] accessor.cc:621: Received notification for node id = 4172ea067fbd59a05ab1840b5ce95992093c2ba6ce015b84e244c04f, IsAlive = 0
[2022-02-18 12:45:43,852 I 149 149] accessor.cc:621: Received notification for node id = 24a402550a9f823c5c9363781ded45fdbe12b1a1583acf1e39d692e5, IsAlive = 0
[2022-02-18 12:45:43,852 I 149 149] accessor.cc:621: Received notification for node id = 6d1e9e8b77b37a6bb67a5c42329731c92bfa9c3e105a99f6821d6879, IsAlive = 0
[2022-02-18 12:45:43,852 I 149 149] accessor.cc:621: Received notification for node id = b76495e7ca48a8cda8f549549e3a29faf9d043f02d568a7bec327b0d, IsAlive = 0
[2022-02-18 12:45:43,852 I 149 149] accessor.cc:621: Received notification for node id = f56954eb046cf9d0e4ee05202f65950d2ad96323f0a85aa016f9d510, IsAlive = 0
[2022-02-18 12:45:43,852 I 149 149] accessor.cc:621: Received notification for node id = 37222c6670db1283ceb8d920c8b4b67ca88365827db3b260ca989e58, IsAlive = 0
[2022-02-18 12:45:43,852 I 149 149] accessor.cc:621: Received notification for node id = aeb3bc318d4d344024e3f46617e28cb92500c886d7986c7054439294, IsAlive = 0
[2022-02-18 12:45:43,852 I 149 149] accessor.cc:621: Received notification for node id = 2be5ef0b55d358ec781959abc42583053369951e9c16348692fa5acd, IsAlive = 0
[2022-02-18 12:45:44,011 C 149 149] node_manager.cc:809: [Timeout] Exiting because this node manager has mistakenly been marked as dead by the GCS: GCS didn't receive heartbeats from this node for 30000 ms. This is likely because the machine or raylet has become overloaded.
*** StackTrace Information ***
    ray::SpdLogMessage::Flush()
    ray::RayLog::~RayLog()
    ray::raylet::NodeManager::NodeRemoved()
    ray::gcs::NodeInfoAccessor::HandleNotification()
    std::_Function_handler<>::_M_invoke()
    ray::gcs::GcsPubSub::ExecuteCommandIfPossible()::{lambda()#1}::operator()()
    std::_Function_handler<>::_M_invoke()
    std::_Function_handler<>::_M_invoke()
    boost::asio::detail::completion_handler<>::do_complete()
    boost::asio::detail::scheduler::do_run_one()
    boost::asio::detail::scheduler::run()
    boost::asio::io_context::run()
    main
    __libc_start_main
```

dashboard console log:
```
TypeError: Cannot read properties of undefined (reading 'length')
    at NodeInfo.tsx:211:52
    at Array.filter (<anonymous>)
    at _n (NodeInfo.tsx:211:30)
    at Xi (react-dom.production.min.js:153:146)
    at Go (react-dom.production.min.js:175:309)
    at ys (react-dom.production.min.js:263:406)
    at El (react-dom.production.min.js:246:265)
    at fl (react-dom.production.min.js:246:194)
    at sl (react-dom.production.min.js:239:172)
    at react-dom.production.min.js:123:115
```

dashboard chrome debug info, nodes variable:
```
actors: {}
errorCount: 0
logCount: 15018
logUrl: "http://None:34568/logs"
raylet:
metricsExportPort: 64298
nodeId: "bb94a38968cfbbe03c89ec021254a28ef8d26d03500506c7e40099b2"
nodeManagerAddress: "192.168.11.27"
nodeManagerHostname: "GPU-1127"
nodeManagerPort: 41588
numWorkers: 1
objectManagerPort: 41424
objectStoreAvailableMemory: 4294967296
objectStoreSocketName: "/tmp/ray/session_2022-02-11_21-11-04_446422_248/sockets/plasma_store.1"
objectStoreUsedMemory: 0
rayletSocketName: "/tmp/ray/session_2022-02-11_21-11-04_446422_248/sockets/raylet.1"
resourcesTotal: {}
state: "ALIVE"
storeStats: {objectStoreBytesAvail: '4294967296'}
timestamp: "0"
viewData: (44) [{…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}, {…}]
[[Prototype]]: Object
workers: []
[[Prototype]]: Object
```

**Comment by xychu** [comment]
Ray 1.10 same issue.
From dashboard side, the cause of `TypeError: Cannot read properties of undefined (reading 'length')` is the response of `nodes?view=details` seems contain raylet `state: "ALIVE"` client but without `gpu` and a lot more real ALIVE client should have.

Bad
`
456: {raylet: {storeStats: {objectStoreBytesAvail: "10286394163"},…}, actors: {}, workers: [],…}
actors: {}
errorCount: 0
logCount: 27808
raylet: {storeStats: {objectStoreBytesAvail: "10286394163"},…}
metricsExportPort: 54766
nodeId: "416ffdcb376ea39e217ed2af5358fc441e2e4134bcc9d8247fc33af8"
nodeManagerAddress: "10.177.86.72"
nodeManagerHostname: "cluster-ray-worker-2dnf6"
nodeManagerPort: 46591
objectManagerPort: 41043
objectStoreAvailableMemory: 10286394163
objectStoreSocketName: "/tmp/ray/session_2022-04-19_22-59-55_204733_1518036/sockets/plasma_store"
objectStoreUsedMemory: 0
rayletSocketName: "/tmp/ray/session_2022-04-19_22-59-55_204733_1518036/sockets/raylet"
resourcesTotal: {}
state: "ALIVE"
storeStats: {objectStoreBytesAvail: "10286394163"}
timestamp: "0"
viewData: [{viewName: "process_startup_time_ms",…}, {viewName: "object_directory_updates",…},…]
workers: []
` 
vs Good
`
455: {raylet: {storeStats: {objectStoreBytesAvail: "10285737984"},…},…}
actors: {}
bootTime: 1650464007.43
cmdline: ["/home/ray/anaconda3/lib/python3.8/site-packages/ray/core/src/ray/raylet/raylet",…]
cpu: 0.3
cpus: [4, 4]
disk: {/: {total: 1, used: 0, free: 1, percent: 0}}
errorCount: 0
gpus: []
hostname: "cluster-ray-worker-zxrxq"
ip: "10.177.34.73"
loadAvg: [[3.02, 2.3, 1.46], [0.76, 0.57, 0.36]]
logCount: 27856
logUrl: "http://10.177.34.73:42756/logs"
mem: [34359738368, 31692296192, 7.8, 2667442176]
network: [10082403366, 3069663853]
networkSpeed: [24242.882829328224, 33839.24524360485]
now: 1650503787.124332
raylet: {storeStats: {objectStoreBytesAvail: "10285737984"},…}
metricsExportPort: 57637
nodeId: "56e7dd30522dd9c242cf48feb2d606ee043d5397a5265283e927d827"
nodeManagerAddress: "10.177.34.73"
nodeManagerHostname: "cluster-ray-worker-zxrxq"
nodeManagerPort: 35898
objectManagerPort: 41161
objectStoreAvailableMemory: 10285737984
objectStoreSocketName: "/tmp/ray/session_2022-04-19_22-59-55_204733_1518036/sockets/plasma_store"
objectStoreUsedMemory: 0
rayletSocketName: "/tmp/ray/session_2022-04-19_22-59-55_204733_1518036/sockets/raylet"
resourcesTotal: {}
state: "ALIVE"
storeStats: {objectStoreBytesAvail: "10285737984"}
timestamp: "0"
viewData: [{viewName: "process_startup_time_ms",…}, {viewName: "heartbeat_report_ms",…},…]
workers: []
`

**Comment by xychu** [comment]
Seems the dashboard crash may have been fixed in https://github.com/ray-project/ray/pull/21650
But still not sure if it's still a problem that some node raylet `status:ALIVE` but without resource usage from `ReporterAgent`.
Also the dispath remote task failures mentioned by @newmanwang  https://github.com/ray-project/ray/issues/18889#issuecomment-1041134711

> Ray dispatch all new remote task to this node, and all these task failed with:
> 
> `ray.exceptions.RuntimeEnvSetupError: The runtime_env failed to be set up.`

**Comment by scottsun94** [comment]
@xychu @newmanwang @yyzlong @witt3rd @spolcyn @zhaichenhao1889 @jakestrouse00 @Murphy-OrangeMud 
In ray 2.0, we made the previous experimental dashboard the default dashboard.
Not sure if you have upgraded to 2.0, but are you still running into this issue with this new dashboard?

**Comment by richardliaw** [comment]
This should be addressed by the new dashboard changes in 2.0 and 2.1.

Let's close this for now and open a new thread if the error persists.
